tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,218 @@
+
+    select_console('svirt');
+
+    # Guest Power Off
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Power Off');
+        esxi_vm_power_off($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("! ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Power On
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered off/) {
+        record_info('Guest Power On');
+        esxi_vm_power_on($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Reset
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Reset');
+        esxi_vm_power_reset($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Restart
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Restart');
+        esxi_vm_power_reboot($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Shutdown
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Shutdown');
+        esxi_vm_power_shutdown($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("! ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+}
+
+sub networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered off/) {
+        esxi_vm_power_on($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    reset_consoles;
+    select_console("sut");
+    assert_screen("grub2", 90);
+    send_key 'ret';
+    select_console("root-console", timeout => 180);
+
+    assert_script_run("ping -c3 openqa.suse.de");
+    assert_script_run("ping -c3 www.suse.com");
+}
+
+sub clock_sync_tests {
+    my ($vm_name, $vm_id, $vm_ip) = @_;
+    my ($g_datetime, $h_datetime, $g_difftime);
+
+    # Enable timesync service in guest VM
+    if (script_output("vmware-toolbox-cmd timesync status", proceed_on_failure => 1, timeout => 90) == "Disabled") {
+        $g_datetime = script_output("date -u +'\%Y-\%m-\%d \%H:\%M:\%S'");
+        record_info('Guest current time before syncing up', $g_datetime);
+        assert_script_run("vmware-toolbox-cmd timesync enable", timeout => 90);
+        $h_datetime = get_host_timestamp();
+        record_info('Host time', $h_datetime);
+        $g_datetime = script_output("date -u +'\%Y-\%m-\%d \%H:\%M:\%S'");
```

Comment:
Is it correct that this function should be executed directly on the vm's console? It seems in code you have switched to "select_console("sut");" in networking_tests, so here the code to switch to sut console is not given, right? If yes, I'd suggest to give a comment for it, to let people more easily understand the test code.

An even better way is to decouple the close relationship(or env dependency) between network test and clock test, so that they can be more flexible.

Reply:
Yes, it's the open-vm-tools function within the guest.  Maybe adding a judgment to check the current console before test running can decouple the relationship between each tests.

Reply:
Added following statements to ensure the test is under the correct console.
`    record_info('Clock Sync Tests');
     select_console("sut") if (current_console() != "sut");
`